### ISSUE TITLE
Int type is now "firmly" 32 bits wide -- do we want that?

### ISSUE THREAD

**Issue by andyarvanitis** [purescript/purescript#1251]
Since `Int` has an instance of `Bounded` in the prelude now, it's effectively defined to be 32 bits wide:
```purescript
instance boundedInt :: Bounded Int where
  top = 2147483647
  bottom = -2147483648
```
I'm fine with this, but maybe we want it to be backend/platform-defined instead, with these bounds provided through FFI values?

I think I'm pretty much ok either way, just planning ahead for some native C++14 backend libraries.

**Comment by garyb** [comment]
Originally I didn't add a `Bounded` instance for `Int` for this exact reason, but I guess we forgot later on. :wink:

Providing the values through the FFI seems like a good way of working around this to me. What do you think @paf31?

**Comment by paf31** [comment]
That sounds good, but it seems unfortunate to have to put such a low level thing with `Bounded`. Seems like it would belong in `Prim`, but obviously we don't define values there.

**Comment by andyarvanitis** [comment]
Ok, I'll put something together and submit a Prelude PR sometime soon. I don't need it right away, so if you think of any arguments for keeping the bounds explicit we can reconsider the change.

**Comment by hdgarrood** [comment]
Another approach could be to stick with a fixed width regardless of backend, and rename the current `Int` to `Int32`. That does at least avoid ugliness with `Prim`. I think Rust took a similar approach (although probably for different reasons - I don't know what those were).

**Comment by andyarvanitis** [comment]
Ultimately, it is backend-dependent, right? (even for javascript). You could make it explicitly 32-bits (with or without suffix), but then I think you would technically have to enforce it, in case javascript ever changes their bitwise operator behavior (even if that's unlikely).

**Comment by hdgarrood** [comment]
Where are you imagining that it would be enforced?

**Comment by andyarvanitis** [comment]
Dunno -- I think that would be very ugly. Basically I'm just saying that you want the bounds to remain true, and not end up getting values outside of them if something in the backend changes.

I guess the point is less relevant for javascript unless you're able to *programmatically* get those bounds via the FFI (you can in C++). If not, you would have to update the bounds manually anyway. Again, it's probably not a major concern for JS, but in languages like C++ you have more variations and options when mapping types.

**Comment by andyarvanitis** [comment]
Implemented in purescript-prelude pull #42.
